draft-ietf-quic-recovery.md

```diff
@@ -398,8 +417,80 @@ A packet sent on an RTO alarm MUST NOT be blocked by the sender's congestion
 controller. A sender MUST however count these bytes as additional bytes in
 flight, since this packet adds network load without establishing packet loss.
 
+## Multiple Packet Number Space Optimizations {#optimizations}
+
+There are cases where one may be able to gain recovery information from
+acknowledgements of packets in another packet number space, but they rely
+on complex assumptions about the peer’s processing and acknowledgement
+algorithms.  Even those are unable to quickly recover from cases such as
+losing the client's Initial, but receiving the 0-RTT packets.  Below are
+three different optimizations in increasing complexity that minimize
+handshake latency.
+
+### Empty ACK frames
+
+The EMPTY_ACK frame enables faster recovery of lost Initial and Handshake
+packets when the only other outstanding packets are undecryptable.
+In that case, the peer can’t decrypt the packet or packet number,
+but they do know a packet for a QUIC connection of a supported version was
+received.
+
+The receiver SHOULD send an EMPTY_ACK frame soon (e.g., 1ms) after
+undecryptable packets are received, even if those received packets are
+not buffered for later decryption.  The small delay allows for cases when
+0-RTT packets are reordered in front of the Initial, which is not uncommon
+on networks that prioritize small packets.  The receiver should limit the
+number of EMPTY_ACK frames sent to one per packet number space per RTT.
```

Comment:
For 0-RTT packets, we don't have an RTT estimate. Maybe it's better to limit the amount of empty acks to a small number (2?) per encryption level. 2 is the number we use for TLPs (with the argument that one packet might be lost).

Reply:
Fair point about RTT estimate.  We could say limit to 1 until you have an RTT estimate even?